src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldHaveCorrectSuffix.cs

```diff
@@ -83,98 +100,100 @@ private static void AnalyzeCompilationStart(CompilationStartAnalysisContext cont
                     {
-                        saContext.ReportDiagnostic(namedTypeSymbol.CreateDiagnostic(rule, namedTypeSymbol.ToDisplayString(), typeSuffixInfo.Suffix));
+                        context.ReportDiagnostic(namedTypeSymbol.CreateDiagnostic(rule, namedTypeSymbol.ToDisplayString(), typesSuffixes[0],
+                            string.Join("', '", typesSuffixes.Skip(1))));
                     }
-
-                    return;
                 }
-
-                var interfaces = excludeIndirectBaseTypes
-                    ? namedTypeSymbol.Interfaces
-                    : namedTypeSymbol.AllInterfaces;
-
-                if (TryGetTypeSuffix(interfaces, interfaceTypeSuffixMap, userTypeSuffixMap, out var interfaceSuffixInfo) &&
-                    !namedTypeSymbol.Name.EndsWith(interfaceSuffixInfo.Suffix, StringComparison.Ordinal))
+                else
                 {
-                    saContext.ReportDiagnostic(namedTypeSymbol.CreateDiagnostic(DefaultRule, namedTypeSymbol.ToDisplayString(), interfaceSuffixInfo.Suffix));
+                    var interfaces = excludeIndirectBaseTypes
+                        ? namedTypeSymbol.Interfaces
+                        : namedTypeSymbol.AllInterfaces;
+
+                    if (TryGetTypeSuffix(interfaces, interfaceTypeSuffixMap, userTypeSuffixMap, out var interfaceSuffixes, out rule) &&
+                        !interfaceSuffixes.Any(suffix => namedTypeSymbol.Name.EndsWith(suffix, StringComparison.Ordinal)))
+                    {
+                        context.ReportDiagnostic(namedTypeSymbol.CreateDiagnostic(rule, namedTypeSymbol.ToDisplayString(), interfaceSuffixes[0],
+                            string.Join("', '", interfaceSuffixes.Skip(1))));
+                    }
                 }
             }
             , SymbolKind.NamedType);
 
             var eventArgsType = wellKnownTypeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemEventArgs);
-            if (eventArgsType != null)
+            if (eventArgsType == null)
             {
-                context.RegisterSymbolAction((saContext) =>
-                {
-                    const string eventHandlerString = "EventHandler";
-                    var eventSymbol = (IEventSymbol)saContext.Symbol;
-                    if (!eventSymbol.Type.Name.EndsWith(eventHandlerString, StringComparison.Ordinal) &&
-                        eventSymbol.Type.IsInSource() &&
-                        eventSymbol.Type.TypeKind == TypeKind.Delegate &&
-                        ((INamedTypeSymbol)eventSymbol.Type).DelegateInvokeMethod?.HasEventHandlerSignature(eventArgsType) == true)
-                    {
-                        saContext.ReportDiagnostic(eventSymbol.CreateDiagnostic(DefaultRule, eventSymbol.Type.Name, eventHandlerString));
-                    }
-                },
-                SymbolKind.Event);
+                return;
             }
+
+            context.RegisterSymbolAction((saContext) =>
```

Comment:
Rename `saContext` to `context`?
```suggestion
            context.RegisterSymbolAction(context =>
```